params/version.go

```diff
@@ -60,5 +65,28 @@ func VersionWithCommit(gitCommit string) string {
 	if len(gitCommit) >= 8 {
 		vsn += "-" + gitCommit[:8]
 	}
+	if VersionMeta != "stable" {
+		d := getCommitDate(gitCommit)
+		if d != "" {
+			vsn += "-" + d
+		}
+	}
 	return vsn
 }
+
+func getCommitDate(commit string) string {
+	if commit != "" {
+		out, err := exec.Command("git", "show", "-s", "--format=%ct", commit).CombinedOutput()
+		if err != nil {
+			log.Println("Could not get gitCommit date: " + string(out))
+			return ""
+		}
+		ti, err := strconv.ParseInt(strings.TrimSpace(string(out)), 10, 64)
+		if err != nil {
+			log.Println("Could not convert gitCommit date. Parsed timestamp is: " + err.Error())
+			return ""
+		}
+		return time.Unix(ti, 0).Format("20060102")
+	}
+	return ""
+}
```

Comment:
This solution would require the entire git repository to be available during Geth's execution. You **need** to do this during build time when the source code is available.

Reply:
Yeah, just noticed this when I was running that it was being pulled during the call. Give me a few minutes...

Reply:
okey dokey, updated. this solution is a bit more involved but it appears this is what you're looking for. let me know!